ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/TrailingCommaRule.kt

```diff
@@ -192,6 +193,34 @@ public class TrailingCommaRule :
         node.reportAndCorrectTrailingCommaNodeBefore(inspectNode, emit, autoCorrect)
     }
 
+    private fun visitClassEntry(
+        node: ASTNode,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+        autoCorrect: Boolean
+    ) {
+        val psi = node.psi
+        require(psi is KtClass)
+
+        if (psi.isEnum()) {
+            val classBody = node.children().singleOrNull { it.psi is KtClassBody }
+                ?: return // Nothing to do for empty enum class
+
+            val lastRelevantChild = classBody.psi.children.toList().asReversed()
+                .first()
+
+            val fullAST = psi
+            val fullPsi = classBody.psi
+
+            if (lastRelevantChild !is KtEnumEntry) {
```

Comment:
If the list of enum entries ends with with a ";" followed by newline, what is then the lastChild? Is it an enum entry or something else which would result in an unwanted early exit.